src/tools/r2rdump/R2RMethod.cs

```diff
@@ -0,0 +1,267 @@
+                }
+                else if (handle.Kind == HandleKind.TypeReference)
+                {
+                    var typeRef = mdReader.GetTypeReference((TypeReferenceHandle)handle);
+                    ClassName = mdReader.GetString(typeRef.Name);
+                }
+            }
+        }
+
+        public override string ToString()
+        {
+            StringBuilder sb = new StringBuilder();
+            if (SignatureTypeCode == SignatureTypeCode.ByReference)
+            {
+                sb.Append("ref ");
+            }
+            sb.AppendFormat($"{ClassName}");
+            if (IsArray)
+            {
+                sb.Append("[]");
+            }
+            return sb.ToString();
+        }
+
+
+    }
+
+    struct RuntimeFunction
+    {
+        /// <summary>
+        /// The relative virtual address to the start of the code block
+        /// </summary>
+        public int EntryPoint { get; set; }
+
+        /// <summary>
+        /// The size of the code block in bytes
+        /// </summary>
+        /// /// <remarks>
+        /// The EndAddress field in the runtime functions section is conditional on machine type
+        /// Size is -1 for images without the EndAddress field
+        /// </remarks>
+        public int Size { get; set; }
+
+        /// <summary>
+        /// The relative virtual address to the unwind info
+        /// </summary>
+        public int UnwindRVA { get; set; }
+
+        public RuntimeFunction(int startRva, int endRva, int unwindRva)
+        {
+            EntryPoint = startRva;
+            Size = endRva - startRva;
+            if (endRva == -1)
+                Size = -1;
+            UnwindRVA = unwindRva;
+        }
+    }
+
+    class R2RMethod
+    {
+        private const int _blockSize = 16;
+
+        public string Name { get; }
+        public SignatureType ReturnType { get; }
+        public SignatureType[] ArgTypes { get; }
+        public int Token { get; }
+
+        public RuntimeFunction NativeCode { get; }
+
+        public R2RMethod(byte[] image, RuntimeFunction[] runtimeFunctions, ref MetadataReader mdReader, MethodDefinitionHandle methodDefHandle, int methodDefEntryPointsOffset)
+        {
+            var methodDef = mdReader.GetMethodDefinition(methodDefHandle);
+
+            BlobReader signatureReader = mdReader.GetBlobReader(methodDef.Signature);
+            SignatureHeader header = signatureReader.ReadSignatureHeader();
+            Name = mdReader.GetString(methodDef.Name);
+            int argCount = signatureReader.ReadCompressedInteger();
+            ReturnType = new SignatureType(ref signatureReader, ref mdReader);
+            ArgTypes = new SignatureType[argCount];
+            for (int i = 0; i < argCount; i++)
+            {
+                ArgTypes[i] = new SignatureType(ref signatureReader, ref mdReader);
+            }
+
+            Token = MetadataTokens.GetToken(methodDefHandle);
```

Comment:
I'm nervous that this will not map correctly. Can you please iterate over `READYTORUN_SECTION_METHODDEF_ENTRYPOINTS` and use the index of that able as the rid to map into the `RUNTIME_FUNCTIONS` table?